week2/address_book/addressbook_create_content/2_project_setup.md

```diff
@@ -0,0 +1,282 @@
+  }
+```
+
+Let's talk a bit about what's going on with that script
+
+```bash
+superstatic -p 3000 2> /dev/null & cucumber-js; lsof -ti tcp:3000 | xargs kill
+```
+
+This will do 3 things when executed:
+
+1. `superstatic -p 3000` starts a web server on port `3000` from the current directory using `superstatic`. Our application will then be available on `http://localhost:3000`  `2> /dev/null &` will silence all logs from the superstatic process and push that to the background to allow the next command to be executed in the current terminal session.
+2. `cucumber-js` will execute the tests from all `.feature` files in our features directory
+3. `lsof -ti tcp:3000 | xargs kill` This last command will kill the web server we started earlier on once the tests are done.
+
+Your `package.json` should now look something like this:
+
+```javascript
+{
+  "name": "address-book",
+  "version": "1.0.0",
+  "description": "Address book application",
+  "main": "index.js",
+  "author": "John Doe <john@acmecorp.tld>",
+  "license": "MIT",
+  "devDependencies": {
+    "cucumber": "^4.2.1",
+    "mocha": "^5.0.5",
+    "puppeteer": "^1.2.0",
+    "superstatic": "^5.0.1"
+  },
+  "scripts": {
+    "cucumber":
+      "superstatic -p 3000 2> /dev/null & cucumber-js; lsof -ti tcp:3000 | xargs kill"
+  }
+}
+```
+
+Now that we have the cucumber script in place, let's go ahead and run it.
+
+```bash
+$ yarn run cucumber
+```
+
+You should get an output similar to the following screenshot on your terminal
+
+![Undefined step-definitions for cucumber](https://raw.githubusercontent.com/CraftAcademy/ca-course-week-2/master/.gitbook/assets/screenshot-2018-04-11-06.58.05.png)
+
+The way cucumber translate the steps written in plain english into actual executable code that tests our application is through what's called step definitions. In the previous output it's telling us that we haven't defined any yet for the steps we have in `main.feature` 
+
+We will copy the suggested steps into a step definition file and run cucumber again. Create a file name `basic-steps.js` inside of the `step-definitions` directory in the `features` directory and paste the following in it.
+
+```javascript
+const { Given, Then } = require("cucumber");
+
+Given("I visit the site", function() {
+  // Write code here that turns the phrase above into concrete actions
+  return "pending";
+});
+
+Then("I should see {string}", function(string) {
+  // Write code here that turns the phrase above into concrete actions
+  return "pending";
+});
+```
+
+If you run your test again now, you should get the following output
+
+![First step in pending state](https://github.com/CraftAcademy/ca-course-week-2/raw/master/.gitbook/assets/screenshot-2018-04-11-07.06.56.png)
+
+Cucumber now tells us that the step `Given I visit the site` is `Pending`. This is because all our step definition does at this stage is return the word `Pending`. We will return to this later on and write some code to get it to pass.
```

Comment:
'This is because what all of our step definitions do, at this stage, is to return the word `Pending`'